giskard/testing/tests/data_quality.py

```diff
@@ -0,0 +1,299 @@
+        return TestResult(passed=False, metric_name="consistency", metric=0, messages=message)
+
+    return TestResult(passed=True, metric_name="consistency", metric=1)
+
+@test(name="Mislabeled Data Test")
+def mislabel(dataset: Dataset, labelled_column: str, reference_columns: Iterable[str]):
+    """
+    Test for detecting mislabelled data.
+
+    Args:
+        dataset (giskard.Dataset): The dataset to test.
+        labelled_column (str): The column containing the labels.
+        reference_columns (Iterable[str]): The columns containing the data to check for consistency.
+
+    Returns:
+        TestResult: The result of the test, containing the indices of the mislabelled data.
+    """
+    # Copy the dataset to avoid changing the original data
+    dataset_copy = dataset.df.copy()
+
+    # Encode the categorical data
+    le = LabelEncoder()
+    for column in dataset_copy.columns:
+        if dataset_copy[column].dtype == 'object':
+            dataset_copy[column] = le.fit_transform(dataset_copy[column])
+
+    # Prepare the data
+    x = dataset_copy[list(reference_columns)]
+    y = dataset_copy[labelled_column]
+
+    # Combine the data and labels
+    data = pd.concat([x, y], axis=1)
+
+    # Train the Isolation Forest model
+    model = IsolationForest(contamination=0.1)
+    model.fit(data)
+
+    # Predict the anomalies in the data
+    anomalies = model.predict(data) == -1
+
+    # Check if any of the anomalies have different labels
+    mislabelled_data = dataset.df[anomalies]
+
+    if not mislabelled_data.empty:
+        message = f"Mislabelled data found: \n{mislabelled_data}"
+        return TestResult(passed=False, metric_name="consistency", metric=0, messages=message)
+
+    return TestResult(passed=True, metric_name="consistency", metric=1)
+
+@test(name="Feature Importance Test")
+def feature_importance(dataset: Dataset, target_column: str, feature_columns: Iterable[str]):
+    """
+    Test for evaluating the importance of each feature to the target variable.
+
+    Args:
+        dataset (giskard.Dataset): The dataset to test.
+        target_column (str): The column containing the target variable.
+        feature_columns (Iterable[str]): The columns containing the features.
+
+    Returns:
+        TestResult: The result of the test, containing the feature importances.
+        The more value the more important a feature is with respect to target variable.
+    """
+    # Prepare the data
+    features = list(feature_columns)
+    x = dataset.df[features]
+    y = dataset.df[target_column]
+    # Train the Random Forest model
+    model = RandomForestClassifier()
+    model.fit(x, y)
+    # Get the feature importances
+    importances = model.feature_importances_
+    feature_importances = dict(zip(features, importances))
+    # Create a message containing the feature importances
+    message = f"Feature importances: \n{feature_importances}"
+
+    return TestResult(passed=True, metric_name="feature_importance",
```

Comment:
Same here, we should implement some kind of logic in order to decide whenever the test pass or fail

Reply:
@kevinmessiaen Could you re-assign this PR to me? I would then continue with the work.